lib/manager/composer/utils.ts

```diff
@@ -73,13 +79,14 @@ export function extractContraints(
   }
   // check last used composer version
   else if (lockParsed?.['plugin-api-version']) {
-    const major = api.getMajor(lockParsed?.['plugin-api-version']);
-    res.composer = `${major}.*`;
+    res.composer = lockParsed?.['plugin-api-version'];
   }
   // check composer api dependency
   else if (composerJson.require?.['composer-runtime-api']) {
     const major = api.getMajor(composerJson.require?.['composer-runtime-api']);
-    res.composer = `${major}.*`;
+    const minor =
+      api.getMinor(composerJson.require?.['composer-runtime-api']) || '*';
+    res.composer = `${major}.${minor}`;
```

Comment:
```suggestion
    res.composer = `^${major}.${minor}`;
```

this allows newer composer versions

Reply:
Thx, did this in https://github.com/renovatebot/renovate/pull/13187